// reactcss/lib/hover.js
'use strict';
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var react_1 = require('react');
exports.hover = function (Component) {
    return (function (_super) {
        __extends(Hover, _super);
        function Hover() {
            var _this = this;
            _super.call(this);
            this.handleMouseOver = function () {
                _this.setState({ hover: true });
            };
            this.handleMouseOut = function () {
                _this.setState({ hover: false });
            };
            this.state = {
                hover: false,
            };
        }
        Hover.prototype.render = function () {
            return onMouseOver = { this: .handleMouseOver };
            onMouseOut = { this: .handleMouseOut }
                >
                    { this: .props };
            {
                this.state;
            }
            />
                < /div>;
        };
        return Hover;
    }(react_1.default.Component));
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = exports.hover;
